Core/src/org/sleuthkit/autopsy/communications/snapshot/CommSnapShotReportWriter.java

```diff
@@ -0,0 +1,126 @@
+
+    }
+
+    /**
+     * Generate and write the html page that shows the snapshot and the state of
+     * the CommunicationFilters
+     *
+     * @throws IOException If there is a problem writing the html file to disk.
+     */
+    @Override
+    protected void writeSnapShotHTMLFile() throws IOException {
+        SimpleDateFormat formatter = new SimpleDateFormat("MMMMM dd, yyyy");
+
+        ImageIO.write(image, "png", getReportFolderPath().resolve("snapshot.png").toFile()); //NON-NLS
+
+        //make a map of context objects to resolve template paramaters against
+        HashMap<String, Object> snapShotContext = new HashMap<>();
+        snapShotContext.put("reportTitle", getReportName()); //NON-NLS
+
+        List<SubFilter> filters = filter.getAndFilters();
+
+        for (SubFilter f : filters) {
+            if (f.getClass().getName().equals(DateRangeFilter.class.getName())) {
+                long startDate = ((DateRangeFilter) f).getStartDate();
+                long endDate = ((DateRangeFilter) f).getEndDate();
+
+                if (startDate > 0) {
+
+                    snapShotContext.put("startTime", formatter.format(new Date((Instant.ofEpochSecond(startDate)).toEpochMilli()))); //NON-NLS
+                }
+
+                if (endDate > 0) {
+                    snapShotContext.put("endTime", formatter.format(new Date((Instant.ofEpochSecond(endDate)).toEpochMilli()))); //NON-NLS
+                }
+            } else if (f.getClass().getName().equals(AccountTypeFilter.class.getName())) {
+                snapShotContext.put("emailSelected", "checked"); //NON-NLS
+
+                Set<Account.Type> types = ((AccountTypeFilter) f).getAccountTypes();
+                for (Account.Type type : types) {
+                    if (type == Account.Type.DEVICE) {
+                        snapShotContext.put("deviceSelected", "checked"); //NON-NLS
+                    } else if (type == Account.Type.PHONE) {
+                        snapShotContext.put("phoneSelected", "checked"); //NON-NLS
+                    } else if (type == Account.Type.EMAIL) {
+                        snapShotContext.put("emailSelected", "checked"); //NON-NLS
+                    } else if (type == Account.Type.FACEBOOK) {
+                        snapShotContext.put("facebookSelected", "checked"); //NON-NLS
+                    } else if (type == Account.Type.TWITTER) {
+                        snapShotContext.put("twitterSelected", "checked"); //NON-NLS
+                    } else if (type == Account.Type.INSTAGRAM) {
+                        snapShotContext.put("instagramSelected", "checked"); //NON-NLS
+                    } else if (type == Account.Type.WHATSAPP) {
+                        snapShotContext.put("whatsAppSelected", "checked"); //NON-NLS
+                    } else if (type == Account.Type.WEBSITE) {
+                        snapShotContext.put("websiteSelected", "checked"); //NON-NLS
+                    }
+                }
+            }
```

Comment:
what about the device filter?

Reply:
We talked yesterday at the standup and Brian didn't think the report needed anything more than the picture.  At this point i had already done the other filters so it was decided to leave as is.  Also the device is in the picture.

Reply:
If Brian is cool with it, then so be it.

Reply:
As a result of a conversation with Richard, I have sent a message to both Richard and Brian to confirm whether or not the table is even needed.

Reply:
My two cents is that is is necessary to understand the image.  Otherwise it could easily be missinterpreted to show the abscence of communications/accounts, when really they are just filtered out.  Maybe Brian figures that context can be provided by analysts in other ways, but I feel negligent to not include it in the report.  Users can always use just the image anyways...